classes/calculation_helper.php

```diff
@@ -67,4 +67,83 @@ public static function div($divident, $divisor) {
 
         return $divisor != 0 ? round($divident / $divisor, 2) : 0.00;
     }
+
+    /**
+     * Aggregates the time spend for all or just one student in the course. The calculation is done in the DB
+     * and the retuned array is can be provided to the plugins timeoverview and activity.
+     *
+     * @param int $courseid
+     * @param int $startdate
+     * @param int $enddate
+     * @param int $userid
+     * @return array
+     * @throws \coding_exception
+     * @throws \dml_exception
+     */
+    public static function get_activity_aggregation($courseid, $startdate, $enddate, $userid = null) {
```

Comment:
get used to defining argument type definitions, because in future php versions that will be required.